### DISCUSSION TITLE
Cannot simulate waverforms of any Pulseq protocol in IDEA

### DISCUSSION THREAD

**Discussion by markymarkymark** [pulseq/pulseq#61] [Q&A]
I have Pulseq working in VE12U, including running various .seq files on the scanner.
In IDEA I cannot get the simulator to run for any *.pro made from POET with the pulseq sequence. POET **does** work for the pulseq binary, and it allows me to load various *.seq files from the Special tab. But all attempts to run _sim -p *.pro_ fail. 

I get this  pop-up window:
![sim_error](https://github.com/user-attachments/assets/59ce47f3-e924-44ce-9697-145fefcf1452)

Here is the result from trying the gre_example.seq that comes with Pulseq:

```
[VE12U-CS-d] Z:\n4\pkg\MrServers\MrImaging\seq\pulseq>sim -p gre_example.pro
17:28:42 INFO  IdeaRun.IdeaSim No Dsv options specified, using values from cfg
17:28:42 INFO  IdeaRun.SeqSimulation Used Dsv options for simulation:
17:28:42 INFO  IdeaRun.SeqSimulation Specified file is a protocol
17:28:42 INFO  IdeaRun.SeqSimulation Protocol file for simulation: gre_example.pro
17:28:42 INFO  IdeaRun.SeqSimulation Run simulation..
17:28:42 INFO  IdeaRun.SeqSimulation Exec: cmd /c MriSeqSimDSPClientd gre_example.pro
!!! some components failed at configureHW, check logfile
   TALESK2284


MriSeqSimDSPClientD.exe   (Build: Feb 17 2021) - V03.01
Simulator DLL:   4.0 - debug version (Build: Feb 17 2021)

This simulator provides the following options:
GPO GPP GPN AD2 NC2 D0X D0Y D0Z GSR GAM M0X M0Y M0Z M1X M1Y M1Z MXX
MXY MXZ PXA PYA PZA PXB PYB PZB PXC PYC PZC SFX SFY SFZ SAP SLL SLT
RFP BON BA0 BB0 BC0 BOS BD0 BE0 BF0 PRF

[System type: dVStar]
[....................................] 0.0%, errors: 0, time: 2024/10/23-17:28:55.188926 4524 MriSeqSimDSPClientd.exe 5068 Thread-5068 -1 Info 0 MrPMU
Srv:PMUSerIF PMUSerIF.cpp:120 PMUServerIF::PMUServerIF PMUSerIF::PMUServerIF Interface Testmode is ON
[....................................] 0.0%, errors: 0, time: Wed Oct 23 17:28:55.550 *** PULSEQ: SeqInit finished normally. Message source (\n4\pkg\M
rServers\MrImaging\seq\pulseq/Arbitrary.cpp:734)
Errors: 0  ()                                               :

   Z:\n4\pkg\MrServers\MrImaging\seq\pulseq\gre_example_ERR.dsv

   Created by DSP-Simulator / V4.0 - debug version (Build: Feb 17 2021)
   17:28:54 on Wednesday, October 23, 2024


= Starting protocol simulation.
  - The protocol contains 1 event blocks.
  - These event blocks do supply DSP instructions for approximately 0.000000.

# ERROR: Sequence run
  - Could not run sequence

= Simulation successfully complete
- Simulation starting duration is 0 us.
- Simulation end duration is 2783138807 us.
17:28:57 INFO  IdeaRun.SeqSimulation Checking for simulation results..
17:28:57 INFO  IdeaRun.SeqSimulation Show simulation results..
17:28:57 INFO  IdeaRun.IdeaVis Exec(async): cmd /c MriSeqSimViewer.exe gre_example_CON.dsv
```

**Comment by m-a-x-i-m-z** [comment]
Hi Mark, 

strange... can you simulate the default protocol (running sim without any parameters?) 

also, did you try to run the sim tool directly from POET? 

Best regards, 
Maxim

**Reply by markymarkymark** [reply]
Yes, I get the same error from POET, and also when running _sim_ w/o arguments. No issue with other sequences in my VE12U IDEA environment (both product and custom).
I also get this pop-up:
![sim_error2](https://github.com/user-attachments/assets/ea97561f-ca08-4b4c-a160-f95beabb9580)

I checked and both those files exist and are readable from within the IDEA shell. **Very strange**.

I also checked Pulseq in my VE11C VM and everything works just fine.
I think there is something peculiar in my VE12U install. Will try installing a clean copy

**Comment by m-a-x-i-m-z** [comment]
Hi Mark, 

please make sure you are working with the most recent release of the Pulseq interpreter (from July 2024). This version is the first one to have a built-in sequence (FID followed by three beeps, one for each gradient axis). If you cannot simulate that one you better to check your IDEA installation ;-)

**Reply by markymarkymark** [reply]
Thanks, Maxim, for the replies. I am running the latest version w/ the built-in seq.
Will let this discussion know if I resolve the issue.

Mark.
